### PR TITLE
fix(node-http-handler): handle NodeHttp2Handler session failure

### PR DESCRIPTION
**Pull request by TysonAndre**
### Issue
#1525 

### Description
detect errors on the NodeHttp2Handler, immediately destroy
connections on unexpected error mode, and reconnect.

Prior to this PR, if the server sent the client a GOAWAY frame,
the session would not be removed from the connection pool and requests
would fail indefinitely. This fixes that.

This tries to avoid keeping streams(requests) on the Http2Session
(tcp connection) from being stuck in an open state waiting for a gentle
close even if there were unexpected protocol or connection errors
during the request, assuming http2 errors are rare.
(if a server or load balancer or network is misbehaving,
close() might get stuck waiting for requests to finish,
especially if requests and sessions don't have timeouts?)

I'm only slightly familiar with http/2 client implementations from
working on clients for Apple Push Notification Service.

- In those, the client could rely on a session and request timeout
  existing, so close() would finish. In aws-sdk-js-v3, timeouts are optional.
- I've seen some strange race conditions prior to node 12 in different
  client implementations for (certain combinations of) close and/or destroying

### Testing
Unit tests were added to confirm that disconnections initiated by the server
resulted in the connection being removed from the connection pool.

More thorough manual tests would be useful to test all possible networking edge cases,
e.g. completely losing a connection to aws load balancers, a connection closing without a goaway frame, etc,
testing this with a high volume of traffic to see if it recovers properly, etc.,
but I'm not familiar with how to simulate that accurately in node

### Additional context
Add any other context about the PR here.

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by trivikr — APPROVED** [review]
LGTM

File: packages/node-http-handler/src/node-http2-handler.spec.ts [hunk 1]
```diff
@@ -121,55 +121,51 @@ describe("NodeHttp2Handler", () => {
...
+        const buffers = [];
+        resultReader.on("data", (chunk) => buffers.push(chunk));
+        resultReader.on("end", () => {
+          resolve(Buffer.concat(buffers).toString("utf8"));
         });
-        expect(responseBody).toBe("test");
-        expect(establishedConnections).toBe(4);
-        expect(numRequests).toBe(5);
-      } finally {
-        mockH2Server3.close();
```
Comment (TysonAndre) [inline packages/node-http-handler/src/node-http2-handler.spec.ts]: The intent of this check was to avoid making other tests fail if this test failed (if they ever reused the same port in the future) - it could waste time debugging if `"handles connections destroyed by servers"` below failed because it's also using port3 (or something even less straightforward).

Now that I think of it, checking for the mockH2Server in an after() handler instead and calling mockH2Server.close() may have been cleaner than this try{}finally{} approach. Thoughts?

Comment (trivikr) [reply packages/node-http-handler/src/node-http2-handler.spec.ts]: `mockH2Server.close()` is called in `afterAll` as mockH2Server is reused between tests.

If any additional servers are created by tests, they need to be closed within the test.
For example, `mockH2Server2` while testing new authority and `mockH2Server3` for goaway frames or connections destroyed by servers.
